R/groupingsets.R

```diff
@@ -29,6 +29,26 @@ cube.data.table = function(x, j, by, .SDcols, id = FALSE, label = NULL, ...) {
     stopf("Argument 'id' must be a logical scalar.")
   if (missing(j))
     stopf("Argument 'j' is required")
+  #implementing NSE in cube
+  jj = substitute(j)
+  usesSD = any(all.vars(jj) == ".SD")
+  if (usesSD) {
+    if (missing(.SDcols)) {
+      .SDcols = names(x)[vapply(x, is.numeric, logical(1L))]
```

Comment:
How do you know SDcols needs to be numeric, for example there are aggregate functions that works on character.